src/umap.cpp

```diff
@@ -36,7 +36,7 @@ static long page_size;
 class umap_page;
 class _umap {
     public:
-        _umap(void* _mmap_addr, size_t _mmap_length, int num_backing_file, umap_backing_file* backing_files);
+        _umap(void* _mmap_addr, size_t _mmap_length,vector<umap_backing_file> *backing_files);
```

Comment:
The constructor should not be changed.  The creation of the vector should be done inside of the constructor.  Please revert constructor back to:
```C++
_umap(void* _mmap_addr, size_t _mmap_length, int num_backing_file, umap_backing_file* backing_files);